docs/proposed/grid-manager/managed-grid.rst

```diff
@@ -0,0 +1,338 @@
+    daemonize tahoe -d storage0 run
+    daemonize tahoe -d storage1 run
+
+.. _skip_ahead:
+
+We can now tell the Grid Manager about our new storage servers::
+
+    tahoe grid-manager --config ./gm0 add storage0 $(cat storage0/node.pubkey)
+    tahoe grid-manager --config ./gm0 add storage1 $(cat storage1/node.pubkey)
+
+To produce a new certificate for each node, we do this::
+
+    tahoe grid-manager --config ./gm0 sign storage0 > ./storage0/gridmanager.cert
+    tahoe grid-manager --config ./gm0 sign storage1 > ./storage1/gridmanager.cert
+
+Now, we want our storage servers to actually announce these
+certificates into the grid. We do this by adding some configuration
+(in ``tahoe.cfg``)::
+
+    [storage]
+    grid_management = true
+
+    [grid_manager_certificates]
+    default = gridmanager.cert
+
+Add the above bit to each node's ``tahoe.cfg`` and re-start the
+storage nodes.
+
+Now try adding a new storage server ``storage2``. This client can join
+the grid just fine, and announce itself to the Introducer as providing
+storage::
+
+    tahoe create-node --introducer $(cat introducer/private/introducer.furl) --nickname storage2 --webport 6301 --webport 6302 --location tcp:localhost:6303 --port 6303 ./storage2
+    daemonize tahoe -d storage2 run
+
+At this point any client will upload to any of these three
+storage-servers. Make a client "alice" and try!
+
+::
+
+    tahoe create-client --introducer $(cat introducer/private/introducer.furl) --nickname alice --webport 6301 --shares-total=3 --shares-needed=2 --shares-happy=3 ./alice
+    daemonize tahoe -d alice run
+    tahoe -d alice mkdir  # prints out a dir-cap
+    find storage2/storage/shares  # confirm storage2 has a share
+
+Now we want to make Alice only upload to the storage servers that the
+grid-manager has given certificates to (``storage0`` and
+``storage1``). We need the grid-manager's public key to put in Alice's
+configuration::
+
+    tahoe grid-manager --config ./gm0 public-identity
+
+Put the key printed out above into Alice's ``tahoe.cfg`` in section
+``client``::
+
+    [grid_managers]
+    example_name = pub-v0-vqimc4s5eflwajttsofisp5st566dbq36xnpp4siz57ufdavpvlq
+
+
+DECIDE:
+ - should the grid-manager be identified by a certificate? exarkun
+   points out: --name seems like the hint of the beginning of a
+   use-case for certificates rather than bare public keys?).
+ - (note the "--name" thing came from a former version of this
+   proposal that used CLI commands to add the public-keys -- but the
+   point remains, if there's to be metadata associated with "grid
+   managers" maybe they should be certificates..)
+
+Now, re-start the "alice" client. Since we made Alice's parameters
+require 3 storage servers to be reachable (``--happy=3``), all their
+uploads should now fail (so ``tahoe mkdir`` will fail) because they
+won't use storage2 and thus can't "achieve happiness".
```

Comment:
hmmmm, i have repeated this on the re-based version of this branch .. either 'mkdir' doesn't follow servers of happiness or there's something deeper wrong?

Reply:
Arrrrgggg .. looks like maybe `mkdir` / directories don't honour happiness? (The integration test for this uses `tahoe put` and `tahoe put` *does* fail if I follow the example but replace any `tahoe mkdir` with `tahoe put`.

For now, I'll change the example to use `put` because the mkdir thing is just its own problem :/

Reply:
ah, right, directories are mutable and https://tahoe-lafs.org/trac/tahoe-lafs/ticket/2060